desk/app/channels-server.hoon

```diff
@@ -842,4 +1004,287 @@
+  ++  give  |=(=gift:agent:gall ho-core(cor (^give gift)))
+  ++  ho-abet
+    %_  cor
+        hooks.hooks
+      ?:(gone (~(del by hooks.hooks) id) (~(put by hooks.hooks) id hook))
+    ==
+  ::
+  ++  ho-abed
+    |=  i=id:h
+    ho-core(id i, hook (~(got by hooks.hooks) i))
+  ::
+  ++  ho-add
+    |=  [name=@t src=@t]
+    ^+  ho-core
+    =.  id
+      =+  i=(end 7 eny.bowl)
+      |-(?:((~(has by hooks.hooks) i) $(i +(i)) i))
+    =/  result=(each vase tang)
+      (compile:utils src)
+    =/  compiled
+      ?:  ?=(%| -.result)
+        ((slog 'compilation result:' p.result) ~)
+      `p.result
+    =.  hook  [id %0 name *data:m src compiled !>(~) ~]
+    =.  cor
+      =/  error=(unit tang)
+        ?:(?=(%& -.result) ~ `p.result)
+      (give-hook-response [%set id name src meta.hook error])
+    ho-core
+  ++  ho-edit
+    |=  [name=(unit @t) src=(unit @t) meta=(unit data:m)]
+    =?  src.hook  ?=(^ src)  u.src
+    =/  result=(each vase tang)
+      (compile:utils src.hook)
+    ?:  ?=(%| -.result)
+      =.  cor
+        %-  give-hook-response
+        [%set id name.hook src.hook meta.hook `p.result]
+      ho-core
+    =?  name.hook  ?=(^ name)  u.name
+    =?  meta.hook  ?=(^ meta)  u.meta
+    =.  compiled.hook  `p.result
+    =.  cor
+      %-  give-hook-response
+      [%set id name.hook src.hook meta.hook ~]
+    ho-core
+  ::
+  ++  ho-del
+    =.  gone  &
+    =.  cor
+      %+  roll
+        ~(tap by (~(gut by crons.hooks) id *(map origin:h cron:h)))
+      |=  [[=origin:h =cron:h] cr=_cor]
+      (unschedule-cron:cr origin cron)
+    =.  crons.hooks  (~(del by crons.hooks) id)
+    =.  order.hooks
+      %+  roll
+        ~(tap by order.hooks)
+      |=  [[=nest:c ids=(list id:h)] or=(map nest:c (list id:h))]
+      =-  (~(put by or) nest -)
+      (skip ids |=(i=id:h =(id i)))
+    =.  delayed.hooks
+      %+  roll
+        ~(tap by delayed.hooks)
+      |=  [[=delay-id:h d=[* delayed-hook:h]] dh=_delayed.hooks]
+      ?.  =(id hook.d)  dh
+      (~(del by dh) delay-id)
+    =.  cor  (give-hook-response [%gone id])
+    ho-core
+  ++  ho-configure
+    |=  [=nest:c =config:h]
+    ^+  ho-core
+    =.  config.hook  (~(put by config.hook) nest config)
+    =.  cor  (give-hook-response [%config id nest config])
+    ho-core
+  ++  ho-wait
+    |=  [=origin:h schedule=$@(@dr schedule:h) =config:h]
+    ^+  ho-core
+    =/  schedule
+      ?:  ?=(@ schedule)  [now.bowl schedule]
+      schedule
+    =/  crons  (~(gut by crons.hooks) id *(map origin:h cron:h))
+    =/  =cron:h  [id schedule config]
+    =.  crons.hooks
+      =-  (~(put by crons.hooks) id.hook -)
+      (~(put by crons) origin cron)
+    =.  cor  (schedule-cron origin cron)
+    =.  cor  (give-hook-response [%wait id origin schedule config])
+    ho-core
+  ++  ho-rest
+    |=  =origin:h
+    ^+  ho-core
+    =/  crons  (~(got by crons.hooks) id)
+    =/  cron  (~(got by crons) origin)
+    =.  crons.hooks
```

Comment:
The usage of `crons` and `cron` here and elsewhere make this quite confusing. Normally getting an item out of a collection should make it singular. One alternative would be to replace `crons` with `crontab`. Then what is now `crons.hooks` would become `crontabs.hooks`, and an item inside that would be `crontab`, which is a collection of `cron` items. 

Another way to do that (and avoid the not-so-nice _crontab_) would be to employ the names `job` (`cron-job`), `cron`, `crons`.